parsers/OPENNEM.py

```diff
@@ -0,0 +1,142 @@
+        y, w, d = df_start.isocalendar()
+        iso_week = "{0}W{1:02d}".format(y, w)
+        raise Exception('Not tested yet')
+        url = f'http://data.opennem.org.au/power/history/5minute/{region}_{iso_week}.json'
+    else:
+        url = f'https://data.opennem.org.au/v3/stats/au/{region.upper()}/power/7d.json'
+
+    # Fetches the last week of data
+    r = (session or requests).get(url)
+    r.raise_for_status()
+    datasets = r.json()['data']
+    df = pd.concat([dataset_to_df(x) for x in datasets], axis=1)
+    # SOLAR_ROOFTOP is only given at 30 min interval, so let's interpolate it
+    df['SOLAR_ROOFTOP'] = df['SOLAR_ROOFTOP'].interpolate(limit=5)
+    return df
+
+
+def sum_vector(pd_series, keys, transform=lambda x: x):
+    # Only consider keys that are in the pd_series
+    filtered_keys = [k for k in keys if k in pd_series.index]
+    if filtered_keys and all([not np.isnan(pd_series[k]) for k in filtered_keys]):
+        return pd_series[filtered_keys].apply(transform).sum()
+    else:
+        return None
+
+
+def fetch_production(zone_key=None, session=None, target_datetime=None, logger=logging.getLogger(__name__)):
+    df = fetch_main_df(zone_key, session, target_datetime, logger)
+    # Index(['DISTILLATE', 'WIND', 'SOLAR_UTILITY', 'BIOENERGY_BIOGAS', 'GAS_OCGT',
+    #    'SOLAR_ROOFTOP', 'GAS_CCGT', 'COAL_BLACK', 'PRICE', 'TEMPERATURE',
+    #    'PRICE', 'COAL_BROWN', 'GAS_STEAM', 'GAS_RECIP'],
+    #   dtype='object')
+    # Make sure charging is counted positively
+    # and discharging negetively
+    if 'BATTERY_DISCHARGING' in df.columns:
+        df['BATTERY_DISCHARGING'] = df['BATTERY_DISCHARGING'] * -1
+
+    objs = [{
+        'datetime': arrow.get(datetime.to_pydatetime()).datetime,
+        'production': {  # Unit is MW
+            'coal': sum_vector(row, ['COAL_BLACK', 'COAL_BROWN']),
+            'gas': sum_vector(row, ['GAS_CCGT', 'GAS_OCGT', 'GAS_RECIP', 'GAS_STEAM']),
+            'oil': sum_vector(row, ['DISTILLATE']),
+            'hydro': sum_vector(row, ['HYDRO']),
+            'wind': sum_vector(row, ['WIND']),
+            'biomass': sum_vector(row, ['BIOENERGY_BIOGAS']),
+            # We here assume all rooftop solar is fed to the grid
+            # This assumption should be checked and we should here only report
+            # grid-level generation
+            'solar': sum_vector(row, ['SOLAR_UTILITY', 'SOLAR_ROOFTOP']),
+        },
+        'storage': {
+            # opennem reports charging as negative, we here should report as positive
+            # Note: we made the sign switch before, so we can just sum them up
+            'battery': sum_vector(row, ['BATTERY_DISCHARGING', 'BATTERY_CHARGING']),
+            # opennem reports pumping as positive, we here should report as positive
+            'hydro': sum_vector(row, ['PUMPS']),
+        },
+        'source': SOURCE,
+        'zoneKey': zone_key,
+    } for datetime, row in df.iterrows()]
```

Comment:
Maybe avoid using `datetime` since it's a module import, how about `dt` instead?